OpenSim/Simulation/Model/ContactGeometry.cpp

```diff
@@ -81,90 +74,110 @@ void ContactGeometry::constructProperties()
     constructProperty_color(defaultColor);
 }
 
-const Vec3& ContactGeometry::getLocation() const
+void ContactGeometry::constructConnectors()
 {
-    return get_location();
+    constructConnector<PhysicalFrame>("frame");
 }
 
+const Vec3& ContactGeometry::getLocation() const
+{ return get_location(); }
+
 void ContactGeometry::setLocation(const Vec3& location)
-{
-    set_location(location);
-}
+{ set_location(location); }
 
 const Vec3& ContactGeometry::getOrientation() const
-{
-    return get_orientation();
-}
+{ return get_orientation(); }
 
 void ContactGeometry::setOrientation(const Vec3& orientation)
-{
-    set_orientation(orientation);
-}
+{ set_orientation(orientation); }
 
-SimTK::Transform ContactGeometry::getTransform()
+SimTK::Transform ContactGeometry::findTransformInBaseFrame() const
 {
-    return SimTK::Transform(Rotation(SimTK::BodyRotationSequence,
-        get_orientation()[0], SimTK::XAxis,
-        get_orientation()[1], SimTK::YAxis,
-        get_orientation()[2], SimTK::ZAxis), get_location());
+    // B: base
+    // F: frame specified in the connector for this object.
+    // P: the (imaginary) frame that is defined (relative to F) by the location
```

Comment:
I don't really like my choice of "imaginary" but is the frame intermediate? It's a leaf. The important thing I want to convey is that this is not an actual OpenSim `Frame` object; it's just a frame defined by the `location` and `orientation` properties. I'll try to make this clearer.